src/components/GroupSystems/GroupImmutableSystems.js

```diff
@@ -0,0 +1,367 @@
+          reloadData={() => {
+            if (calculateSelected() > 0) {
+              dispatch(selectEntity(-1, false));
+            }
+
+            inventory.current.onRefreshData({}, false, true);
+          }}
+        />
+      )}
+      {updateDevice && (
+        <AsyncComponent
+          appName="edge"
+          module="./UpdateDeviceModal"
+          navigateProp={useNavigate}
+          locationProp={useLocation}
+          notificationProp={notificationProp}
+          paramsProp={useParams}
+          updateModal={updateModal}
+          setUpdateModal={setUpdateModal}
+          refreshTable={() => true}
+        />
+      )}
+      {!addToGroupModalOpen && (
+        <InventoryTable
+          columns={(columns) => mergeColumns(columns)}
+          hideFilters={{ hostGroupFilter: true }}
+          // getEntities={entities}
+          getEntities={customGetEntities}
+          tableProps={{
+            isStickyHeader: true,
+            variant: TableVariant.compact,
+            canSelectAll: false,
+            actionResolver: (row) => [
+              {
+                title: (
+                  <ActionDropdownItem
+                    requiredPermissions={REQUIRED_PERMISSIONS_TO_MODIFY_GROUP(
+                      groupId
+                    )}
+                    noAccessTooltip={NO_MODIFY_GROUP_TOOLTIP_MESSAGE}
+                    onClick={() => {
+                      setCurrentSystem([row]);
+                      setRemoveHostsFromGroupModalOpen(true);
+                    }}
+                  >
+                    Remove from group
+                  </ActionDropdownItem>
+                ),
+                style: {
+                  padding: 0, // custom component creates extra padding space
+                },
+              },
+              {
+                title: (
+                  <ActionDropdownItem
+                    isAriaDisabled={
+                      deviceData && !deviceData.find((obj) => obj === row.id)
+                    }
+                    requiredPermissions={REQUIRED_PERMISSIONS_TO_MODIFY_GROUP(
+                      groupId
+                    )}
+                    noAccessTooltip={NO_MODIFY_GROUP_TOOLTIP_MESSAGE}
+                    onClick={() => {
+                      setCurrentSystem([row]);
+                      navigate(`/insights/inventory/${row.id}/update`);
+                    }}
+                  >
+                    Update
+                  </ActionDropdownItem>
+                ),
+                style: {
+                  padding: 0, // custom component creates extra padding space
+                },
+              },
+            ],
+          }}
+          actionsConfig={{
+            actions: [
+              [
+                <ActionButton
+                  key="add-systems-button"
+                  requiredPermissions={REQUIRED_PERMISSIONS_TO_MODIFY_GROUP(
+                    groupId
+                  )}
```

Comment:
Just asking for clarification: the inventory RBAC permissions do cover the edge systems too? So by saying I have "inventory:groups:read", I am able to read both conventional and immutable systems groups?

Reply:
yes, the rbac is being handled on our backend response as well